saas/api/metrics.py

```diff
@@ -60,6 +64,40 @@ def get(self, request, *args, **kwargs): #pylint: disable=unused-argument
         return Response(result)
 
 
+class RevenueAPIView(ProviderMixin, APIView):
+    """
+    Produce revenue stats
+    """
+    def get(self, request, *args, **kwargs):
+        ends_at = request.GET.get('ends_at', None)
+        if ends_at:
+            ends_at = parse_datetime(ends_at)
+        ends_at = datetime_or_now(ends_at)
+
+        table_key = request.GET['table_key']
```

Comment:
That's true. What would you prefer it to do? It certainly is an error not to provide this parameter.

Would it be better to move it to a positional argument in the request path? In which case, not including it would generate a 404, instead.